e2e/init/init_test.go

```diff
@@ -102,26 +102,20 @@ var _ = Describe("init flow", func() {
 	})
 
 	Context("app logs", func() {
-		var (
-			appLogs    []client.AppLogsOutput
-			appLogsErr error
-		)
-
-		BeforeAll(func() {
-			appLogs, appLogsErr = cli.AppLogs(&client.AppLogsRequest{
-				ProjectName: projectName,
-				AppName:     appName,
-				EnvName:     "test",
-				Since:       "1h",
-			})
-		})
-
-		It("should not return an error", func() {
-			Expect(appLogsErr).NotTo(HaveOccurred())
-		})
 
 		It("should return valid log lines", func() {
-			Expect(len(appLogs)).NotTo(Equal(0))
+			var appLogs []client.AppLogsOutput
+			var appLogsErr error
+			Eventually(func() ([]client.AppLogsOutput, error) {
+				appLogs, appLogsErr = cli.AppLogs(&client.AppLogsRequest{
+					ProjectName: projectName,
+					AppName:     appName,
+					EnvName:     "test",
+					Since:       "1h",
+				})
+				return appLogs, appLogsErr
+			}, "60s", "10s").ShouldNot(BeEmpty())
```

Comment:
What is this magic 🧙‍♂️ how does gomega understand that `appLogs` should not be empty? The function inside `Eventually` returns two things

Reply:
It is a blessed framework 🙏:

> The function that you pass to Eventually can have more than one return value. In that case, Eventually passes the first return value to the matcher and asserts that all other return values are nil or zero-valued. This allows you to use Eventually with functions that return a value and an error – a common pattern in Go. For example, say you have a method on an object named FetchNameFromNetwork() that returns a string value and an error. Given an instance then you could simply write: